design-proposals/export-import/vm-export-import.md

```diff
@@ -0,0 +1,166 @@
+
+## Virtual Machine Archive File Format
+
+A Vritual Machine Archive is a xz compressed tar archive with the following structure
+
+```
+.
+├── manifest
+├── resources
+│   ├── pvc-block-pvc.yaml
+│   ├── pvc-filesystem-pvc.yaml
+│   ├── pvc-virtiofs-pvc.yaml
+│   └── vm.yaml
+└── volumes
+    ├── block-pvc
+    │   └── disk.img
+    ├── filesystem-pvc
+    │   └── disk.img
+    └── virtiofs-pvc
+        ├── file1
+        ├── file2
+        └── file3
+```
+
+### Manifest
+
+The `manifest` file lists all files (excluding itself) in the archive and their SHA-1 hashes.  It can be used to ensure that corrumted data is not imported.
+
+### Resources
+
+The `resources` path contains copies of all the Kubernetes resources associaterd to the Virtual Machine.
+
+### Volumes
+
+The `volumes` path contains an entry for each exported volume.
+
+## VirtualMachineExport API
+
+The VirtualMachineExport API allows users to download a Virtual Machine Archive from a URL.
+
+### User Flow
+
+1.  Create a VirtualMachineExport resource
+2.  Wait for the `Ready` condition of the VirtualMachineExport to be `True`
+3.  Download the Virtual Machine Archive by getting the URL in `status.url`
+
+### Export a VirtualMachine
+
+``` yaml
+apiVersion: export.kubevirt.io/v1alpha1
+kind: VirtualMachineExport
+metadata:
+    name: export-vm1
+spec:
+    source:
+        apiGroup: kubevirt.io
+        kind: VirtualMachine
+        name: vm1
+```
+
+### Export a VirtualMachine
+
+``` yaml
+apiVersion: export.kubevirt.io/v1alpha1
+kind: VirtualMachineExport
+metadata:
+    name: export-snap1
+spec:
+    source:
+        apiGroup: snapshot.kubevirt.io
+        kind: VirtualMachineSnapshot
+        name: snap1
+```
+
+### VirtualMachineExport Status
+
+``` yaml
+apiVersion: export.kubevirt.io/v1alpha1
+kind: VirtualMachineExport
+metadata:
+    namespace: ns1
+    name: export-vm1
+spec:
+    ...
+status:
+    phase: Ready # Pending|Ready|Terminated
+    token: 8fmf94kf
+    url: https://virt-import.kubevirt.my-cluster/ns1/export-vm1/download?token=8fmf94kf
+    cert: <base64 encoded cert>
+    clusterURL: https://virt-import.kubevirt/ns1/export-vm1/download?token=8fmf94kf
+    clusterCert: <base64 encoded cert>
+    conditions:
+    - type: Ready
+      status: True
+      ...
+      reason: Ready to export Virtual Machine
+```
+
+* Cluster must have Ingress/Route support for `url` and `cert` to be set
+
+## virtctl export vm|vmsnapshot
+
+virtctl will be extended to include an `export` command that does everything in the [User Flow](#user-flow) section with the addition of deleting the VirtualMachineExport when complete.
+
+```
+virtctl export vm vm1 vm1.tar.gz
+```
+
+```
+virtctl export vmsnapshot snap1 snap1.tar.gz
+```
+
+## virtctl import vm
+
+virtctl will be extended to include an `import` command that will create a new VirtualMachine based on the the contents of Virtual Machine Archive.
+
+```
+virtctl import vm import1 vm1.tar.gz
+```
+
+### Import Implementation
```

Comment:
Would you add a section referring to the prior work of https://github.com/kubevirt/vm-import-operator#create-ovirt-resource-mappings-that-defines-how-ovirt-resources-are-mapped-to-kubevirt ? It may well be unfit for our purpose here, but let us state this explicitly.